homeassistant/components/sensor/dsmr.py

```diff
@@ -168,4 +178,46 @@ def translate_tariff(value):
         elif value == '0001':
             return 'low'
         else:
-            return None
+            return STATE_UNKNOWN
+
+
+class DerivativeDSMREntity(DSMREntity):
+    """Calculated derivative for values where the DSMR doesn't offer one.
+
+    Gas readings are only reported per hour and don't offer a rate only
+    the current meter reading. This entity converts subsequents readings
+    into a hourly rate.
+    """
+
+    _previous_reading = None
+    _previous_timestamp = None
+    _state = STATE_UNKNOWN
+
+    @property
+    def state(self):
+        """Return current hourly rate, recalculate if needed."""
+        # check if the timestamp for the object differs from the previous one
+        timestamp = self.get_dsmr_object_attr('datetime')
```

Comment:
Calculations of state should happen in an update method. Since this code does not seem to do any I/O, you can move it to an `async_update` coroutine.